library/std/src/panicking.rs

```diff
@@ -296,10 +296,17 @@ fn default_hook(info: &PanicInfo<'_>) {
             }
             Some(BacktraceStyle::Off) => {
                 if FIRST_PANIC.swap(false, Ordering::SeqCst) {
-                    let _ = writeln!(
-                        err,
-                        "note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace"
-                    );
+                    let _ = if let Ok(_) = crate::env::var("PSModulePath") {
+                        writeln!(
+                            err,
+                            "note: run with `$env:RUST_BACKTRACE=1` environment variable to display a backtrace."
+                        )
+                    } else {
+                        writeln!(
+                            err,
+                            "note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace."
+                        )
+                    };
```

Comment:
Because .NET Core has made itself sufficiently cross-platform, my understanding is that for macOS, Linux, or Windows, we may be in PowerShell (and for all I know, PS runs on the BSDs too, with sufficient hackery). So we should not simply cfg this out if we accept this PR. And we can only know the value of this env var at runtime, which means we have to include these two strings both in the binary, and we have to make the branch.

While LLVM may successfully merge these strings into the same one, I think that if we accept this, we are probably better off splitting this write up and only writing `"$env:"` conditionally, with a fairly similar logic for the hint for using `RUST_BACKTRACE="full"`. Which is valid syntax in a POSIX shell, by the way, so we can again reuse most of the same string parts.

Reply:
I feel like the strings are too tiny to have an actual real world effect on performance/binary size? Like, yes, you are correct and I agree that simply inserting "$env:" would be smaller. But it's barely anything in the first place 🤷🏻

Reply:
Well... I agree it's very little, but a very little is a lot if we're running on a machine with only MB, not GB, of RAM, one that nonetheless fits the Linux kernel and... probably someone's copy of the Almquist shell, they're usually lighter than bash. A lot of people in such cases complain about the weight of panic code, especially for uses where they're not interactively running the device and it's just a tiny server for some things.

On the other hand, maybe seeing too many requests on the Rust issue tracker to optimize out Literally Two Bytes from a binary has taken a toll on my sanity and I have finally Lost It. You decide.

Reply:
oof, sounds rough. Fair enough, tbh.

Will commit later 👍🏻